tools/autograd/gen_autograd.py

```diff
@@ -68,6 +66,21 @@
 # returns a view of its inputs
 RETURNS_VIEWS_OF_INPUT = set(VIEW_FUNCTIONS.keys()).union({'chunk', 'narrow'})
 
+# note: some VIEW_FUNCTIONS are returning multiple views. This means that the engine
+# has to forbid inplace operations on their reults. If such function is only doing views
+# and is added to the list below, in the case where an inplace operation is performed on
+# the output, the custom backward defined in derivatives.yaml will be ignored and we
+# will use the view informations to create a valid graph where each output is the output
+# of a separate Node (AsStridedBackward) that has one output each.
+# A function should be added to this list if:
+# - You want to allow inplace modification of the result.
+# - You are aware that inplace modification of the output will change the memory usage of the
+#   backward as the multiple Nodes we create will force the gradient buffer to be created
+#   for each Node, instead of once in the original multi-output Node.
+# A function should NOT be added to this list if:
+# - Your backward function does more than just the backward of the views.
```

Comment:
Is there ever a situation where functions in core would not faithfully represent the backwards?

Reply:
The forward could have some side-effects, not tracked by the autograd, that require a specific backward.

For example a function that promises change in the layout? 
Also, I can see some internal users writing a codegened version of the [Broadcast](https://github.com/pytorch/pytorch/blob/8fd3eaed257884361828217f3aeaabc0296ae906/torch/nn/parallel/_functions.py#L9-L32) we have. It returns a mix of views and not views and does a specific backward because it moves some gradients between GPUs.

Reply:
OK, so put that down in the comment :)